### PR TITLE
Integrate HD44780 into Screen HIL

### PR DESCRIPTION
**Pull request by teonaseverin**
### Pull Request Overview

This pull request integrates the HD44780 LCD screen into the screen HIL. It adds the TEXT pixel format, as well as a new function to the `hil::screen::ScreenSetup` trait: `fn screen_command()`. This function can be used to send specific driver commands to the device, that cannot be sent through the API exported by the screen HIL. Such commands, for example for HD44780, can be clearing everything written on the screen, making the cursor visible on the screen or making the cursor blink. 
By integrating the capsule into the screen HIL, the userspace will comunicate with it only through the screen syscalls and not by using the hd44780's specific syscalls. Because of that, I removed the Driver implementation from the capsule. 

### Testing Strategy

This pull request was tested using a NUCLEO-F429ZI board. 

### Documentation Updated

- [x] Removed the documentation about the capsule's syscalls from /docs/syscalls/

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: boards/components/src/adc.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    };};
+}
+
+#[macro_export]
+macro_rules! adc_component_helper {
+    ($A:ty) => {{
+        use capsules::virtual_adc::AdcUser;
+        use core::mem::MaybeUninit;
+        static mut BUF: MaybeUninit<AdcUser<'static, $A>> = MaybeUninit::uninit();
+        &mut BUF
```
Comment (alistair23) [inline boards/components/src/adc.rs]: This ADC work should be in a separate PR

File: boards/nucleo_f429zi/Makefile [hunk 2]
```diff
@@ -3,6 +3,9 @@
 TARGET=thumbv7em-none-eabi
 PLATFORM=nucleo_f429zi
 
+APP=../../../libtock-c/examples/hd44780/build/cortex-m4/cortex-m4.tbf
+KERNEL=$(TOCK_ROOT_DIRECTORY)/target/$(TARGET)/debug/$(PLATFORM).elf
+KERNEL_WITH_APP=$(TOCK_ROOT_DIRECTORY)/target/$(TARGET)/debug/$(PLATFORM)-app.elf
```
Comment (alistair23) [inline boards/nucleo_f429zi/Makefile]: Why add this back?

File: boards/components/src/adc.rs [hunk 3]
```diff
@@ -1,81 +0,0 @@
...
-
-#[macro_export]
-macro_rules! adc_component_helper {
-    ($A:ty) => {{
-        use capsules::virtual_adc::AdcUser;
-        use core::mem::MaybeUninit;
-        static mut BUF: MaybeUninit<AdcUser<'static, $A>> = MaybeUninit::uninit();
-        &mut BUF
-    };};
-}
```
Comment (alistair23) [inline boards/components/src/adc.rs]: Don't add this just to remove it later.

**Comment by teonaseverin** [comment]
Sorry for the mix-up, there was a problem with the master on my fork. I rebased it and deleted the unnecessary commits.
